bulk-load/src/main/java/no/sikt/nva/data/report/api/etl/transformer/BulkTransformerHandler.java

```diff
@@ -0,0 +1,163 @@
+    public static final String NQUADS = ".nquads";
+    public static final ObjectMapper mapper = JsonUtils.dtoObjectMapper;
+    private static final String KEY_BATCHES_BUCKET
+        = ENVIRONMENT.readEnv("KEY_BATCHES_BUCKET");
+    public static final String EVENT_BUS = ENVIRONMENT.readEnv("EVENT_BUS");
+    public static final String TOPIC = ENVIRONMENT.readEnv("TOPIC");
+    public static final String PROCESSING_BATCH_MESSAGE = "Processing batch: {}";
+    public static final String LAST_CONSUMED_BATCH = "Last consumed batch: {}";
+    public static final String LINE_BREAK = "\n";
+
+    private final S3Client s3ResourcesClient;
+    private final S3Client s3BatchesClient;
+    private final S3Client s3OutputClient;
+    private final EventBridgeClient eventBridgeClient;
+
+    // TODO: Default constructor
+    public BulkTransformerHandler(S3Client s3ResourcesClient,
+                                  S3Client s3BatchesClient,
+                                  S3Client s3OutputClient,
+                                  EventBridgeClient eventBridgeClient) {
+        super(KeyBatchRequestEvent.class);
+        this.s3ResourcesClient = s3ResourcesClient;
+        this.s3BatchesClient = s3BatchesClient;
+        this.s3OutputClient = s3OutputClient;
+        this.eventBridgeClient = eventBridgeClient;
+    }
+
+    @Override
+    protected Void processInput(KeyBatchRequestEvent input,
+                                AwsEventBridgeEvent<KeyBatchRequestEvent> event,
+                                Context context) {
+        var startMarker = getStartMarker(input);
+        var location = getLocation(input);
+        var batchResponse = fetchSingleBatch(startMarker);
+
+        var batchKey = batchResponse.contents().getFirst().key();
+        if (Boolean.TRUE.equals(batchResponse.isTruncated())) {
+            sendEvent(constructRequestEntry(batchKey, context, location));
+        }
+
+        var keys = extractContent(batchKey);
+        var nquads = mapToIndexDocuments(keys, location).stream()
+                         .map(this::mapToNquads)
+                         .collect(Collectors.joining(System.lineSeparator()));
+        if (!nquads.isEmpty()) {
+            persistNquads(nquads);
+        }
+        logger.info(LAST_CONSUMED_BATCH, batchResponse.contents().getFirst());
+        return null;
+    }
+
+    private void persistNquads(String nquads) {
+        var request = PutObjectRequest.builder()
+                          .bucket(ENVIRONMENT.readEnv(LOADER_BUCKET))
+                          .key(UUID.randomUUID() + NQUADS)
+                          .build();
+        s3OutputClient.putObject(request, RequestBody.fromString(nquads));
+    }
+
+    private String mapToNquads(IndexDocument content) {
+        // TODO: Refactor GraphName to allow construction of path of GraphName.
+        // This is a silly workaround to make GraphName work.
+        var documentIdentifier = content.getResource().at("/id").toString() + ".nt";
```

Comment:
It looks like you have already implemented this method for `IndexDocument` object:

```
 @JsonIgnore
    public String getDocumentIdentifier() {
        return Optional.ofNullable(consumptionAttributes.getDocumentIdentifier())
            .map(SortableIdentifier::toString)
            .orElseThrow(() -> new RuntimeException(MISSING_IDENTIFIER_IN_RESOURCE));
    }
```

Reply:
It returns the SortableIdentifier…will refactor the entire handling of these two things in a subsequent PR